### ISSUE TITLE
Lack of documentation for CircularReferenceCounter

### ISSUE THREAD

**Issue by maelick** [getkin/kin-openapi#916]
After updating a project with v0.114.0 to the latest, I had a rather complex specification that failed to load because of the default CircularReferenceCounter of 3.

Increasing the value to 42 fixed the issue but I had to dig through the changelog to figure out which change broke my code (it appears it's v0.115.0, most likely #768) and figure out the solution.

It would be nice to have some documentation for it, at least in the code/godoc of the variable, and ideally also in the README (under "Loading OpenAPI document"?)

I don't mind creating a PR for it, but I don't know myself what the variable exactly does, why it defaults to 3 and what are the consequences for increasing it.

**Comment by AnatolyRugalev** [comment]
Circular references should not be a problem after #970, hence, no documentation is required 🙂. `CircularReferenceCounter` has been removed
